### PR TITLE
Upgrade MSBuildLocator to version 1.6.1 

### PR DESCRIPTION
**Pull request by JoeRobich**
This includes several fixes around resolving hostfxr. (See https://github.com/microsoft/MSBuildLocator/pull/230 & https://github.com/microsoft/MSBuildLocator/pull/236)

resolves https://github.com/dotnet/format/issues/1945

### CODE REVIEW

**Comment by mthalman** [comment]
I updated the version of source-build-externals because the previous version contained 1.5.5 of MSBuildLocator. So this update to 1.6.1 cause a source-build prebuilt error to occur. The latest version of source-build-externals has version 1.6.1 so this should resolve the error.

**Comment by mthalman** [comment]
@JoeRobich or @sharwell, could you investigate the build errors? This change is needed to unblock code flow for the sdk: https://github.com/dotnet/installer/pull/17319.

**Comment by mthalman** [comment]
Looks like there are still failures for aspnetcore and efcore.

File: azure-pipelines-integration.yml [hunk 1]
```diff
@@ -23,49 +23,49 @@ jobs:
         _repoName: "dotnet/format"
         _targetSolution: "format.sln"
         _branchName: "main"
-        _sha: "1ec992c6038af9d42499d3ac88c3fd65d0c2f6ed"
+        _sha: "452f6263277533da0af2b557003256d98fbecc27"
```
Comment (sharwell) [inline azure-pipelines-integration.yml:26]: ❔ Why update all of these?

Comment (JoeRobich) [reply azure-pipelines-integration.yml:26]: Habit. Although I see these repos are now on some very *preview* sdk versions

**Review by mthalman — APPROVED** [review]

File: azure-pipelines-integration.yml (hunk 1 above)
Comment (sharwell) [reply azure-pipelines-integration.yml:26]: If updating the dependency broke formatting for the previous SHA, doesn't that mean some users could see dotnet-format break for the same change?

**Comment by mthalman** [comment]
@dotnet-policy-service rerun

**Comment by mthalman** [comment]
@mmitche - Can you help with this CLA issue?

**Comment by mmitche** [comment]
Nope. But I can merge around it!

**Pull request merged**